src/server/game/Spells/SpellEffects.cpp

```diff
@@ -4321,7 +4321,12 @@ void Spell::EffectLeapBack(SpellEffIndex effIndex)
     float speedxy = m_spellInfo->Effects[effIndex].MiscValue / 10.f;
     float speedz = damage/ 10.f;
     //1891: Disengage
-    unitTarget->JumpTo(speedxy, speedz, m_spellInfo->SpellIconID != 1891);
+    //4022: Rolling Throw
+    unitTarget->JumpTo(speedxy, speedz, (m_spellInfo->SpellIconID != 1891 && m_spellInfo->SpellIconID != 4022));
+    if (m_spellInfo->SpellIconID == 4022) 
+    {
+        unitTarget->JumpTo(speedxy, speedz);
```

Comment:
is this doing 2 jumps ? as unitTarget->JumpTo() is called twice

Reply:
yeah using else

Reply:
`    if (m_spellInfo->SpellIconID != 1891) //1891: Disengage
    {
        unitTarget->JumpTo(speedxy, speedz)
    }
    else if (m_spellInfo->SpellIconID == 4022) //4022: Rolling Throw
    {
        unitTarget->JumpTo(speedxy, speedz);
    }`
    
    more or less like this?